04 SFC Login/README.md

```diff
@@ -0,0 +1,1243 @@
+
+## Steps
+
+- `npm install` to install previous sample dependencies:
+
+```
+npm install
+```
+
+- Delete `Hello.vue`.
+
+- Update `App.vue`:
+
+### ./src/App.vue
+
+```diff
+<template>
+  <div>
+    <h1>{{message}}</h1>
+-    <hello-component
+-      :message="message"
+-      :on-change="onChange"
+-    />
+  </div>
+</template>
+
+<script lang="ts">
+import Vue from 'vue';
+- import HelloComponent from './Hello.vue';
+
+export default Vue.extend({
+  name: 'App',
+- components: {
+-   HelloComponent,
+- },
+  data() {
+    return {
+      message: 'Hello from Vue.js',
+    };
+  },
+- methods: {
+-   onChange(event) {
+-     this.message = event.target.value;
+-   }
+- },
+});
+</script>
+
+```
+
+- Update `index.html`:
+
+### ./src/index.html
+
+```diff
+<html lang="en">
+  <head>
+    <meta charset="utf-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+    <title>Vue.js by sample</title>
+  </head>
+  <body>
+-   <div class="well">
+-     <h1>Sample app</h1>
+-   </div>
+    <div id="root">
+    </div>
+  </body>
+</html>
+```
+
+- Create `login` page:
+
+### ./src/pages/login/components/Header.vue
+
+```javascript
+<template>
+  <div class="panel-heading">
+    <h3 class="panel-title">
+      <p>Please sign in</p>
+      <p>login: admin / pwd: test</p>
+    </h3>
+  </div>
+</template>
+
+<script lang="ts">
+import Vue from "vue";
+
+export default Vue.extend({
+  name: 'HeaderComponent',
+});
+</script>
+
+```
+
+### ./src/pages/login/components/Form.vue
+
+```javascript
+<template>
+  <div class="panel-body">
+    <form role="form">
+      <div class="form-group">
+        <input
+          class="form-control"
+          placeholder="e-mail"
+          type="text"
+        />
+      </div>
+      <div class="form-group">
+        <input
+          class="form-control"
+          placeholder="password"
+          type="password"
+        />
+      </div>
+      <button
+        type="button"
+        class="btn btn-lg btn-success btn-block"
+      >
+        Login
+      </button>
+    </form>
+  </div>
+</template>
+
+<script lang="ts">
+import Vue from 'vue';
+
+export default Vue.extend({
+  name: 'FormComponent',
+});
+</script>
+
+```
+
+### ./src/pages/login/components/index.ts
+
+```javascript
+import HeaderComponent from './Header.vue';
+import FormComponent from './Form.vue';
+
+export { HeaderComponent, FormComponent };
+
+```
+
+### ./src/pages/login/Page.vue
+
+```javascript
+<template>
+  <div class="container-fluid">
+    <div class="row">
+      <div class="col-md-4 col-md-offset-4">
+        <div class="panel panel-default">
+          <header-component />
+          <form-component />
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import Vue from 'vue';
+import { HeaderComponent, FormComponent } from './components';
```

Comment:
Is this working? I get this error:

![image](https://user-images.githubusercontent.com/4374977/44151195-18694110-a0a2-11e8-9644-e039e4b97697.png)